python/ray/serve/_private/http_proxy.py

```diff
@@ -322,6 +337,14 @@ async def __call__(self, scope, receive, send):
                 scope, receive, send
             )
 
+        # The current autoscale logic can downscale nodes with ongoing requests if the
+        # node doesn't have replicas and has no object references. This counter and
+        # the dummy object reference will have to keep the node alive while draining
+        # requests, so they are not dropped unintentionally.
+        self._ongoing_requests += 1
+        if self._ongoing_requests > 0 and self._ongoing_requests_dummy_obj_ref is None:
```

Comment:
Could you pull the logic for incrementing/decrementing `self._ongoing_requests` to a separate method? This method is getting pretty long.